### PR TITLE
doc: utilize smartmenus

### PR DESCRIPTION
**Pull request by miri64**
As of version 1.8.12 Doxygen uses [smartmenus](https://www.smartmenus.org/) for the navbar on top. Our design should do so as well (while also supporting <1.8.12 Doxygen output).

Fixes https://github.com/RIOT-OS/RIOT/issues/6910.

### CODE REVIEW

**Comment by gebart** [comment]
Will test this after the weekend.

**Comment by gebart** [comment]
Discovered a minor issue, the arrows for expanding the tree in the side bar are overlapped by the text. 

image

**Comment by miri64** [comment]
> Fixes the problem for me on Doxygen 1.8.13.
> However, was the doc/doxygen/index.html file added by mistake?

Fixed

> Discovered a minor issue, the arrows for expanding the tree in the side bar are overlapped by the text. 
> 
> image

Can you send me the html files again? I did not observe this issue. :-/

**Comment by gebart** [comment]
I managed to locate the problem. It seems like the design update is switching the box model from content-box to border-box, which causes the 16px width arrow span to include the padding, which is set by navtree.js to (16px * indent_level). If we can switch back to the default content-box model then everything should be OK again, I think. I'm not well versed in modern web technologies so I couldn't find where to change it, but there are mentions of border-box when I git grep the doc/doxygen/src tree.

**Comment by gebart** [comment]
Found some more info:
http://stackoverflow.com/questions/18854259/why-did-bootstrap-3-switch-to-box-sizing-border-box

Bootstrap is simply incompatible with Doxygen without patching templates/html/navtree.js in the Doxygen sources.

The navtree looks fine on my side when removing the box-sizing specifiers in bootstrap.min.js, but I don't know if it will have other side-effects on the layout when the dimensions in CSS are not as expected.

**Comment by miri64** [comment]
Actually I already fixed that for the current version at doc.riot.org (that's what [this is about](https://github.com/RIOT-OS/RIOT/blob/master/doc/doxygen/src/css/riot.css#L92-L95)). Please just send me your sources and I'll fix that again.

**Comment by miri64** [comment]
Alternatively, have a look with your browser's "Inspect" function which selector sets the this?

**Comment by gebart** [comment]
The selector is * found in bootstrap.min.js

**Comment by miri64** [comment]
Then let's try this with a compiled result again, if this isn't a problem for you.

**Comment by miri64** [comment]
(which browser are you using?)

**Comment by kaspar030** [comment]
whats the state here?

**Comment by miri64** [comment]
I need @gebart's compiled doc to fix the issue he pointed out.

**Comment by kaspar030** [comment]
Would it make sense to fix this for .04?

**Comment by miri64** [comment]
It does not affect the build doc at doc.riot-os.org at the moment, and even then is just about some broken styling, so I wouldn't give it that much priority.

**Comment by miri64** [comment]
(+ there is no release-based version of doc.riot-os.org anyway).

**Comment by gebart** [comment]
replacing the selector line at https://github.com/RIOT-OS/RIOT/blob/master/doc/doxygen/src/css/riot.css#L92-L95 by
```css
#nav-tree img, #nav-tree span {
```
seem to fix it on my side.

**Comment by miri64** [comment]
Rebased and fixed in `riot.less` (and recompiled `riot.css`)

File: doc/doxygen/src/css/riot.css [hunk 1]
```diff
@@ -202,7 +206,7 @@ div.line.glow {
 }
 span.lineno {
   background-color: transparent;
-  border-right: 3px solid #e3e3e3;
+  border-right: 2px solid #e3e3e3;
```
Comment (miri64) [inline doc/doxygen/src/css/riot.css]: Seems like there was a difference in here ;-)

**Comment by miri64** [comment]
Squashd

**Comment by gebart** [comment]
```
Running './dist/tools/whitespacecheck/check.sh master' x
Command output:

	doc/doxygen/src/css/jquery.smartmenus.bootstrap.css:31: space before tab in indent.
	+ 	border-style: solid;
	doc/doxygen/src/css/jquery.smartmenus.bootstrap.css:123: new blank line at EOF.
	doc/doxygen/src/js/jquery.smartmenus.bootstrap.min.js:4: new blank line at EOF.
	doc/doxygen/src/js/jquery.smartmenus.min.js:4: new blank line at EOF.
	doc/doxygen/src/js/menu.js:30: trailing whitespace.
	+    /* code adapted from original menu.js of Doxygen, but made 
	ERROR: This change introduces new whitespace errors
```

**Comment by miri64** [comment]
Fixed and amended immediately

**Comment by gebart** [comment]
Still one white space error according to Murdock

**Comment by miri64** [comment]
Fixed and amended

**Comment by gebart** [comment]
Murdock is fine, go!

**Pull request merged**